spyder/plugins/editor/widgets/editor.py

```diff
@@ -1774,7 +1776,19 @@ def _write_to_file(self, fileinfo, filename):
         correct encoding without doing any error handling.
         """
         txt = fileinfo.editor.get_text_with_eol()
-        fileinfo.encoding = encoding.write(txt, filename, fileinfo.encoding)
+        try:
+            fileinfo.encoding = encoding.write(txt, filename,

```

Comment:
@steff456, please move the logic you added here directly to the `encoding.write` function because that function is used to save other files too (besides the ones in the editor).